### ISSUE TITLE
Adobe FrameMaker 2020 - Menu and Dialog Box Difficulties

### ISSUE THREAD

**Issue by challengemkr20** [pywinauto/pywinauto#1143]
I apologize for creating an issue here. My hope is that I have overlooked something small.

I am attempting to automate Adobe FrameMaker 2020. 

My colleagues are using FrameMaker 2017 which will stop being supported in January 2022. We are skipping version (2019), but since we are using a custom structured (XML) format and custom-built FrameMaker Development Kit (FDK) extension, I want to ensure everything is works without issue. Therefore, I want to test that our existing documentation will open, validate, export to PDF, interact with our custom extensions, and save correctly.  Once everything is confirmed compatible, I'd like to roll out some bulk changes, such as modifying our template. I hoped to use pywinauto since we have more than 2,000 documents. 

If you are adventurous, Adobe FrameMaker 2020 is available with a 30 day trial at this link:
[https://helpx.adobe.com/framemaker/kb/framemaker-downloads.html](https://helpx.adobe.com/framemaker/kb/framemaker-downloads.html)

## Actual Behavior

- I cannot tell what GUI framework FM2020 was written in (QT?). I cannot tell if it is more compatible with the uia or win32 backend. I see different things with each one. I also seem to get different results with .start() or .connect(). 
- It seems like all Adobe apps use a similar GUI schema (Acrobat, Photoshop, After Effects, Premier, etc.)
- The tutorials and examples don't really seem to apply to a complex application like this.
- The dialogs are very messy. I tried making sense with inspect.exe, py_inspect.py, and spyxx.exe. The tools seem to drastically reorder/regroup the windows depending upon what has been maninpulated in FrameMaker.
- FrameMaker's title text changes to the currently open file.
- I'm having trouble getting the _Application_ class dialogs to connect correctly.
- I cannot get the .menu_select() method to work. I cannot find the control identifiers.
- To work around the missing menu, I tried to use the _Desktop_ class to .click() on it. This then created a PopupMenu which I could navigate using the keyboard. Then I was able to use sendkeys() to enter the filepathname and open the book files.
- I thought I was very clever, until I hit a brick wall: I cannot find the "book" dialog box. This is a problem because I need to open all the .fm files so that they are converted from FM2017 to FM2020 format. Once that is done I can export to PDF.

### Default Control Identifiers - "File", "Edit" "Insert", etc. Missing
Here is the output from my python console.
The problem is that the main program menu bar doesn't have any control identifiers.
pywinauto automatically assigns this as  <instance>.OWLMenu

WARNING: I don't think there is much userful in this first output.
```
from pywinauto import Desktop, Application
app = Application().start(r'C:\Program Files\Adobe\Adobe FrameMaker 2020\FrameMaker.exe')
dlg = app.window(title='Adobe FrameMaker (Structured)')
dlg.print_control_identifiers()
Control Identifiers:
FrameFamily16.MdiFrame - 'Adobe FrameMaker (Structured)'    (L98, T421, R1034, B1521)
['Adobe FrameMaker (Structured)FrameFamily16.MdiFrame', 'FrameFamily16.MdiFrame', 'Adobe FrameMaker (Structured)']
child_window(title="Adobe FrameMaker (Structured)", class_name="FrameFamily16.MdiFrame")
<A huge list truncated of OWLblahblahblah items >
````

From inspect/spy++/py_inspect I found that the menu I'm looking for is "OWLMenuBar", but from this _Application_ instance there aren't any control identifiers. 
```
dlg.OWLMenuBar.print_control_identifiers()
Control Identifiers:
OWL.MenuBar - ''    (L139, T429, R752, B450)
['OWL.MenuBar']
child_window(class_name="OWL.MenuBar")
````

### Desktop Class Instance Attempt
Here is how I used the Desktop class to initially try to work-around my problems.

```
deskapp = Desktop().window(title='Adobe FrameMaker (Structured)') # Removed backend="uia"
deskapp.OWLMenuBar.wait("enabled")
deskapp.OWLMenuBar.draw_outline()
deskapp.OWLMenuBar.click()
deskapp.OWLMenuBar.type_keys("{DOWN}")
```
OWLMenuBar is PopupMenu for each open File, Edit, Insert menu with New, Open, Save, etc. commands.uia works here, but it doesn't work above. I only found this by iterating through through the top "Desktop 1 Pane" and stopping once I found a matching friendlyclassname.

```
cntxt = Desktop(backend="uia")["Context"] 
if cntxt.exists():
    cntxt.wait("enabled")
    cntxt.MenuItem2.click_input() 
```

I swear, I was able to see the list of menu bar options once without clicking, but I could never invoke them using the menuselect commands. I usually get pywinauto.findwindows.ElementNotFoundError or Attribute errors.

I hope there is a kind soul out there who could help me through this mess.

## Specifications

  - Pywinauto version: 0.6.8
  - Python version and bitness: 3.10
  - Platform and OS: Windows 10 (x86; AMD64/64-bit)

**Comment by challengemkr20** [comment]
Update: I figured it out; the trick was using Microsoft Spy++ and the "Find Window" option. I dragged the target onto the OWL.Menu bar and this displayed the handle. 

Once I figured that out, I could use:
app = Application().connect(handle=0xXXXXXXXX)

Then, I would put FrameMaker in various states and use the Python Console in PyCharm and do things like this:
for item in app.window(title=the_filename).children(): 
    print(item)

This would let me manipulate the "tree" like this to find the buttons.
app.windows()[12].children()[61].click()

I still need to see if I can find the OWLMenu.Bar objects, but for now things are working okay and I'm slowly progressing through new pop-ups and situations.

Also, using the handle is not ideal, but I'm sure I'll figure out a better way in the near future as I refactor the program. I may even end up writing a blog about this.
